### DISCUSSION TITLE
Age-Disaggregated Survey Indices with sdmTMB

### DISCUSSION THREAD

**Discussion by niafall** [sdmTMB/sdmTMB#291] [Q&A] [answered]
Hi All;

I have contacted @seananderson via email and he suggested I post here.

I am currently exploring `sdmTMB` as a means of generating survey indices of fish in two separate age classes (i.e. fish aged zero and one years old). My group has been looking at geostatistical mixed-effects models as they have the flexibility to deal with issues in the data, such as incomplete spatial coverage in one year of the survey, while offering flexibility in model structure and the estimation of latent fields. The work is in a fairly early stage, but so far I have been modelling the two age classes entirely separately, and then looking at the consistency between model outputs in terms of cohort tracking after the fact. The initial results show varying levels, moderate levels of correlation, but I was wondering how important is it that the two age classes are currently modelled separately, if at all. 

My initial email asked whether it is possible to "fit an `sdmTMB` model to abundance/density data for two or more age classes, and calculate age-based indices from the output". Of course, age could easily be included as a factor in the model formula, but I am unsure whether the `get_index` function can explicitly separate out the two indices after the fact. I gather there may be functionality in the package to include independent year effects for each age class, I'm just unsure of what the most biologically correct approach is. Should latent fields be fully independent for each age class or autocorrelated? Should the relationship between age-classes be explicitly included in the model, in order to reflect cohort progression through time? Would the results be similar to having entirely separate models?

Apologies for the number of questions, but I'm still very much somewhere along the learning curve for these models.

Many thanks.

**Comment by seananderson** [comment, marked as answer]
There are lots of options here.

The `get_index()` function can certainly separate the two indexes after the fact, and you would do that just by supplying 2 separate `newdata` data frames to the predict function and then the `get_index()` function. Predict first for one age class and then predict for the other age class.

But then the question is how you want to set up the model. The simplest version would share spatial and/or spatiotemporal random effects but perhaps have an interaction between some year variable and the age class variable so that they would have their own means. They could have what is effectively different spatial and/or spatiotemporal random effects by making use of the `spatial_varying` formula and some combination of an age class factor predictor and optionally a year factor predictor. By default, each spatially varying random field has its own SD, but those could be 'mapped' to have the same SD through `sdmTMBcontrol()`. Let me know if there's interest and perhaps we could add a vignette demonstrating this.

The one thing that's not possible is to explicitly model some correlation in the random effects across the age classes. If you want to go that route, then tinyVAST would be a better fit. There's a relevant example at the end of this vignette https://vast-lib.github.io/tinyVAST/articles/web_only/VAST.html

There is also an sdmTMB branch that currently has correlated random walks in it for something like this. It's not fully fleshed out in terms of all the helper functions, but I could point you to that if desired.
